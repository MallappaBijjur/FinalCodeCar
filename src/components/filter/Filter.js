import React, { useState } from 'react';
import '../home-car.css';
import Dropdown from 'react-bootstrap/Dropdown';
import Button from 'react-bootstrap/Button';
import { All_COLORS, ALL_MANUFACTURERS } from '../../constants';

const DropdownCustom = compProps => (
	<Dropdown>
	  <Dropdown.Toggle className="dropdown-white" id="dropdown-basic">
	    {compProps.selected}
	  </Dropdown.Toggle>
	  <Dropdown.Menu>
	  	{
				compProps && compProps.items && compProps.items.map(elem => <Dropdown.Item
																			key={elem}
																			href="#/action-1"
																			onSelect={() => compProps.setSelect(elem, compProps.type)}
																		>
																			{elem}
																		</Dropdown.Item>)
	  	}
	  </Dropdown.Menu>
	</Dropdown>
);

const Filter = props => {
	const { colors, manufacturers, handleFilter } = props;
	const [state, setState] = useState({ selectedColor: All_COLORS, selectedManufacturer: ALL_MANUFACTURERS });
	const setSelect = (elem, type) => {
		setState({...state, [type]: elem})
	}
	let availableCars = colors;
	let names = manufacturers && manufacturers.map(({name}) => name);
	names && names.unshift("All manufacturers");
	availableCars  && availableCars.unshift("All car colors");

	return (	<section className="filter" data-testid="filter-comp">
				<section className="filter-box">
					<p>Color</p>
					<DropdownCustom
						key="color"
						items={colors} 
						selected={state.selectedColor} 
						type={"selectedColor"} 
						setSelect={setSelect} 
					/>
					<p>Manufacturer</p>
					<DropdownCustom
						key="manufacturer"
						items={names}
						selected={state.selectedManufacturer}
						type={"selectedManufacturer"}
						setSelect={setSelect}
					/>
					<Button className="filter-button" data-testid="filter" onClick={() => handleFilter(state)}>Filter</Button>
				</section>
			</section>)
};


export default Filter;