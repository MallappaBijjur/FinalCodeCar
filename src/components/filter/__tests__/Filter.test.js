import React from 'react';
import ReactDOM from 'react-dom';
import { render, cleanup } from '@testing-library/react';
import rerender from 'react-test-renderer';
import Filter from '../Filter';

describe("Filter", () => {

	const originalError = console.error
	beforeAll(() => {
		console.error = (...args) => {
			if (/Warning.*not wrapped in act/.test(args[0])) {
				return
			}
			originalError.call(console, ...args)
		}
	})

	afterAll(() => {
		console.error = originalError
	});

	afterEach(cleanup);
	
	const fakeColorData = {"colors":["red","blue","green","black","yellow","white","silver"]};
	const fakeManufacturersData = {
		"manufacturers":[
			 {
					"name":"Audi",
					"models":[]
			 },
			 {
					"name":"BMW",
					"models":[]
			 },
			 {
					"name":"Chrysler",
					"models":[]
			 }]
			};
    const handleFilter = jest.fn();

	it("renders without crashing", ()=>{
		const div = document.createElement("div");
		ReactDOM.render(<Filter />, div);
	});
	
	it('renders Filter component', () => {
        const { queryByTestId } = render(<Filter
                                        colors={fakeColorData.colors}
                                        manufacturers={fakeManufacturersData.manufacturers}
                                        handleFilter={handleFilter}
                                        />);   
		const linkElement = queryByTestId(/filter-comp/i);
		expect(linkElement).toBeInTheDocument();
	});
	
	it("matches snapshot", ()=> {
		const tree = rerender.create(<Filter
                                        colors={fakeColorData.colors}
                                        manufacturers={fakeManufacturersData.manufacturers}
                                        handleFilter={handleFilter}
                                    />).toJSON();
		expect(tree).toMatchSnapshot();
	});
});




