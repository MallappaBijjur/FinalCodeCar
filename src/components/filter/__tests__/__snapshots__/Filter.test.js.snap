// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter matches snapshot 1`] = `
<section
  className="filter"
  data-testid="filter-comp"
>
  <section
    className="filter-box"
  >
    <p>
      Color
    </p>
    <div
      className="dropdown"
      onKeyDown={[Function]}
    >
      <button
        aria-expanded={false}
        aria-haspopup={true}
        className="dropdown-white dropdown-toggle btn btn-primary"
        disabled={false}
        id="dropdown-basic"
        onClick={[Function]}
        type="button"
      >
        All car colors
      </button>
    </div>
    <p>
      Manufacturer
    </p>
    <div
      className="dropdown"
      onKeyDown={[Function]}
    >
      <button
        aria-expanded={false}
        aria-haspopup={true}
        className="dropdown-white dropdown-toggle btn btn-primary"
        disabled={false}
        id="dropdown-basic"
        onClick={[Function]}
        type="button"
      >
        All manufacturers
      </button>
    </div>
    <button
      className="filter-button btn btn-primary"
      data-testid="filter"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Filter
    </button>
  </section>
</section>
`;
